py/pylya/cf.py

```diff
@@ -93,19 +94,20 @@ def cf(pix):
 def fast_cf(z1,r1,w1,d1,z2,r2,w2,d2,ang,same_half_plate):
     wd1 = d1*w1
     wd2 = d2*w2
-    rp = abs(r1-r2[:,None])*sp.cos(ang/2)
+    if rp_min<0 : rp = (r1-r2[:,None])*sp.cos(ang/2)
```

Comment:
The problem is that the choice of sign of rp_min will change the result at rp>0... it doesn't quite make sense. You could use the x_correlation boolean that you defined earlier, no?